File: Selenium/Activities/Selenium-Java/activities/Activity5.java
/*   Input Events
Using Selenium:

Open a new browser to https://www.training-support.net/selenium/input-events
Get the title of the page and print it to the console.
On the page, perform:
Left click and print the value of the side in the front.
Double click to show a random side and print the number.
Right click and print the value shown on the front of the cube.
Close the browser.

 */

package activities;

import io.github.bonigarcia.wdm.WebDriverManager;
import org.openqa.selenium.By;
import org.openqa.selenium.WebDriver;
import org.openqa.selenium.firefox.FirefoxDriver;
import org.openqa.selenium.interactions.Actions;

public class Activity5 {

    public static void main(String[] args) {

// Set up the Firefox driver(GeckoDriver)
        System.setProperty(FirefoxDriver.SystemProperty.BROWSER_LOGFILE, "/dev/null");
        WebDriverManager.firefoxdriver().setup();
        // Create a new instance of the Firefox driver
        WebDriver driver;
        driver = new FirefoxDriver();

        // Create the Actions object
        Actions builder = new Actions(driver);

        // Open the browser
        driver.get("https://training-support.net/selenium/input-events");
        //1. get the title of webpage
        System.out.println("Title of webpage is" + driver.getTitle());

        //Left click and print the value of the side in the front.
        builder.click().pause(2000).build().perform();
        String frontText = driver.findElement(By.className("active")).getText();
        System.out.println("Value of side: "  + frontText);

        //Double click to show a random side and print the number.
        builder.doubleClick().pause(2000).build().perform();
        String number = driver.findElement(By.className("active")).getText();
        System.out.println( " Random side number is " + number);

        //Right click and print the value shown on the front of the cube.
        builder.contextClick().pause(2000).build().perform();
        String frontValue = driver.findElement(By.className("active")).getText();
        System.out.println("Front Value shown on front of cube is " + frontValue);

        //Close active session
        driver.close();


    }

}

File: Selenium/Activities/Selenium-Python/activities/Activity5.py
"""
Input Events
Using Selenium:

Open a new browser to https://www.training-support.net/selenium/input-events
Get the title of the page and print it to the console.
On the page, perform:
Left click and print the value of the side in the front.
Double click to show a random side and print the number.
Right click and print the value shown on the front of the cube.
Close the browser.
"""
import active as active
from selenium import webdriver
from selenium.webdriver import ActionChains
from selenium.webdriver.common.by import By
from selenium.webdriver.firefox.service import Service as FirefoxService

from webdriver_manager.core import driver
from webdriver_manager.firefox import GeckoDriverManager

# Set up the Firefox Driver with WebDriverManger
service1 = FirefoxService(GeckoDriverManager().install())

# Start the Driver
with webdriver.Firefox(service=service1) as driver:
    # Open the browser to the URL
    driver.get("https://www.training-support.net/selenium/input-events")
    print("Home page Title is ", driver.title)

    # Create the Actions object
    actions = ActionChains(driver)

    # Left click and print the value of the side in the front.
    actions.click().pause(10).perform()
    frontText = driver.find_element(By.CLASS_NAME, "active")
    print("Value of side:",  frontText.text)

    # Double click to show a random side and print the number.
    actions.double_click().pause(10).perform()
    number = driver.find_element(By.CLASS_NAME, "active")
    print(" Random side number is ",  number.text)

    # Right click and print the value shown on the front of the cube.
    actions.context_click().pause(10).perform()
    frontValue = driver.find_element(By.CLASS_NAME, "active")
    print("Front Value shown on front of cube is ",  frontValue.text)

    # Close active session
    driver.quit()
